### PR TITLE
add COCO MAP

### PR DESCRIPTION
**Pull request by Menooker**
## What changes were proposed in this pull request?

Add Mean Average Precision using COCO's algorithm. Add support in MAP for `Table` output and target.

## How was this patch tested?

UT

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/segmentation/MaskUtils.scala [hunk 1]
```diff
@@ -61,29 +61,72 @@ object PolyMasks {
...
+  override def countsLength: Int = _counts.length
+
+  override def get(idx: Int): Long = {
+    MaskUtils.uint2long(_counts(idx))
+  }
+
+  override def counts: Array[Int] = _counts
+}
+
+class RLEMasksFloat(private val _counts: Tensor[Float], height: Int, width: Int)
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/segmentation/MaskUtils.scala]: I don't think we need two types of RLEMask

Comment (Menooker) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/segmentation/MaskUtils.scala]: Changed back to old implementation

**Comment by zhangxiaoli73** [comment]
@Menooker Please merge it if jenkins pass.

**Comment by Menooker** [comment]
jenkins pass http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/989/

**Comment by Menooker** [comment]
> @Menooker Please merge it if jenkins pass.

OK, it's merged!

**Pull request merged**